Fix rendering resetting email on Windows

dmarkowicz opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->

<!--
    Show us you choose the right branch.
    Different branches are used for different things :
    - 4.x is for everything backwards compatible, like patches, features and deprecation notices
    - master is for deprecation removals and other changes that cannot be done without a BC-break
    More details here: https://github.com/sonata-project/SonataUserBundle/blob/4.x/CONTRIBUTING.md#the-base-branch
-->
I am targeting this branch, because it is BC fix.

<!--
    Specify which issues will be fixed/closed.
    Remove it if this is not related.
-->

## Changelog

<!-- MANDATORY
    Fill the changelog part inside the code block.
    Follow this schema: http://keepachangelog.com/
-->

<!-- REMOVE EMPTY SECTIONS -->
```markdown
### Fixed
- Fixed rendering resetting email on Windows
```

## Subject
<!-- Describe your Pull Request content here -->
PHP_EOL (used to spliting template) on Windows does not work for developer who use unix style line endings \n because it expects \r\n in template file what is wrong assumption.
Line endings in template depend on git client config:
- core.autocrlf = true (Checkout Windows-style, commit Unix-style)
- core.autocrlf = input (Checkout as-is, commit Unix-style)

or even developer IDE setup (unix/windows style). This causes the template is not always divided and the email subject contains the entire content of the template.

For spliting lines I used \R group (see https://nikic.github.io/2011/12/10/PCRE-and-newlines.html).

core23 commented on src/Controller/AdminResettingController.php:
```diff
@@ -186,15 +186,31 @@ private function sendResettingEmailMessage(UserInterface $user): void
+    {
         // Render the email, use the first line as the subject, and the rest as the body
-        $renderedLines = explode(PHP_EOL, trim($rendered));
+        $renderedLines = preg_split('/\R/', trim($renderedTemplate), 2, PREG_SPLIT_NO_EMPTY);
```
This looks wrong. Why would you have different line encodings inside your application?

dmarkowicz replied:
Simple test to illustrate my issue on Windows, because PHP_EOL is "\r\n" there.
```php
    public function testEOL()
    {
        $template = "Subject\nBody";

        $renderedLines = explode(PHP_EOL, trim($template));
        $subject = array_shift($renderedLines);
        $body = implode(PHP_EOL, $renderedLines);

        $this->assertEquals('Subject', $subject);
        $this->assertEquals('Body', $body);
    }
```
```
1) x::testEOL
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'Subject'
+'Subject
+Body'
```

OskarStark commented:
How does the Swiftmailer handles something like this?

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

greg0ire commented on tests/Controller/AdminResettingControllerTest.php:
```diff
@@ -29,4 +29,54 @@ public function testItIsInstantiable(): void
+            ->setMethods(['get'])
+            ->getMock();
+        $controller->expects($this->once())->method('get')->with('mailer')
+            ->will($this->returnValue($mailer));
```
You could use https://github.com/sebastianbergmann/phpunit/blob/64d377cc488cd3707f34f46266c8d22e21f1da06/src/Framework/TestCase.php#L1323 here

greg0ire commented on tests/Controller/AdminResettingControllerTest.php:
```diff
@@ -29,4 +29,54 @@ public function testItIsInstantiable(): void
+        $reflectionMethod->setAccessible(true);
+
+        return $reflectionMethod->invokeArgs($object, $args);
+    }
```
Bonus points if you move the code outside the controller and avoid this.

dmarkowicz commented:
Rebased and refactored for last changes. It fixes also unit test in current version on Windows with sources fetched with git config: core.autocrlf = input.
```
There was 1 failure:

1) Sonata\UserBundle\Tests\DependencyInjection\MailerTest::testSendResettingEmailMessage
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'Subject'
+'Subject
+Mail content'

D:\sonata-project\SonataUserBundle\tests\Mailer\MailerTest.php:91
D:\sonata-project\SonataUserBundle\src\Mailer\Mailer.php:78
D:\sonata-project\SonataUserBundle\tests\Mailer\MailerTest.php:97
```
